bundles/http/Citrix/Canvas-LMS/0b9f4444-56d6-4b4e-b108-5624645e71d2.js

```diff
@@ -0,0 +1,410 @@
+        }
+      } else {
+        throw new Error(
+          `Enrollments sync failed ${enrollmentResponse.status}:${enrollmentResponse.statusText}.`
+        );
+      }
+
+      if (announcementResponse.ok) {
+        let announcement = await announcementResponse.json();
+        for (const announcementValue of announcement) {
+          for (const enrollmentValue of enrollment) {
+            const announcementBody = {
+              id: announcementValue.id,
+              enrollme_id: enrollmentValue.id,
+              enrollme_course_id: enrollmentValue.course_id,
+              title: announcementValue.title,
+              message: announcementValue.message,
+              enrollme_user_login_id: enrollmentValue.user.login_id,
+              enrollme_role: enrollmentValue.role,
+              posted_at: announcementValue.posted_at,
+              delayed_post_at: announcementValue.delayed_post_at,
+              author_display_name: announcementValue.author.display_name,
+              url: announcementValue.url,
+            };
+            dataStore.save("announcements", announcementBody);
+          }
+        }
+      } else {
+        throw new Error(
+          `Announcement sync failed ${announcementResponse.status}:${announcementResponse.statusText}.`
+        );
+      }
+      tempEnrollmentLink.forEach((element) => {
+        const enrollmentLink = element.split("; ")[1];
+        if (enrollmentLink == 'rel="next"') {
+          enrollmentPageExists = true;
+        }
+      });
+    } while (enrollmentPageExists);
+
+    tempAnnouncementLink.forEach((element) => {
+      const announcementLink = element.split("; ")[1];
+      if (announcementLink == 'rel="next"') {
+        announcementPageExists = true;
+      }
+    });
+  } while (announcementPageExists);
+}
+
+integration.define({
+  synchronizations: [
+    {
+      name: "canvas",
+      fullSyncFunction: canvasFullSync,
+      incrementalSyncFunction: canvasFullSync,
+    },
+  ],
+  actions: [
+    {
+      name: "Create Course Announcement",
+      parameters: [
+        {
+          name: "title",
+          type: "STRING",
+          required: true,
+        },
+        {
+          name: "message",
+          type: "STRING",
+          required: true,
+        },
+        {
+          name: "courseId",
+          type: "INTEGER",
+          required: true,
+        },
+        {
+          name: "delayed_post_at",
+          type: "DATETIME",
+        },
+        {
+          name: "lock_at",
+          type: "DATETIME",
+        },
+      ],
+      function: createCourseAnnouncement,
+    },
+    {
+      name: "Accept Invitation",
+      parameters: [
+        {
+          name: "courseId",
+          type: "INTEGER",
+          required: true,
+        },
+        {
+          name: "enrollmentId",
+          type: "INTEGER",
+          required: true,
+        },
+        {
+          name: "email",
+          type: "STRING",
+          required: true,

```

Comment:
Why is email parameter here, when it's not used in the action function at all? If you will delete it, don't forget to delete it from the microapp as well.

Reply:
It was used initially but it is not needed now. Agreed and removed the parameter from the script and microapp as well.